packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -227,6 +252,10 @@ export class CanvasManager {
           // canvas is not yet ready... this retry on the next sampling iteration.
           // we don't want to crash the worker if the canvas is not yet rendered.
           if (canvas.width === 0 || canvas.height === 0) {
+            this.debug(canvas, 'not yet ready', {
+              width: canvas.width,
+              height: canvas.height,
+            });
             return;
```

Comment:
we were hitting this code path without clearing the flag in the `snapshotInProgressMap`, meaning that we wouldn't try recording the canvas again